transactions/btc.ts

```diff
@@ -526,6 +527,201 @@ export async function getBtcFeesForNonOrdinalBtcSend(
   }
 }
 
+function getTransactionMetadataForUtxos(
+  recipients: Recipient[],
+  selectedUtxos: UTXO[],
+  changeAddress: string,
+  feeRate: number,
+): TransactionUtxoSelectionMetadata | undefined {
+  const dummyPrivateKey = '0000000000000000000000000000000000000000000000000000000000000001';
+
+  const inputSum = selectedUtxos.reduce<BigNumber>((sum, utxo) => sum.plus(utxo.value), new BigNumber(0));
+  const recipientTotal = recipients.reduce<BigNumber>(
+    (sum, recipient) => sum.plus(recipient.amountSats),
+    new BigNumber(0),
+  );
+
+  // these are conservative estimates
+  const ESTIMATED_VBYTES_PER_OUTPUT = 45; // actually around 50
+  const ESTIMATED_VBYTES_PER_INPUT = 85; // actually around 89 or 90
+  const BITCOIN_DUST_VALUE = 1000;
+
+  const estimatedFees = new BigNumber(feeRate).times(
+    ESTIMATED_VBYTES_PER_OUTPUT * recipients.length + ESTIMATED_VBYTES_PER_INPUT * selectedUtxos.length,
+  );
+  // ensure that the UTXOs can cover the expected outputs
+  if (!inputSum.gt(recipientTotal.plus(estimatedFees))) return undefined;
+
+  {
+    // try transaction with change
+    const tx = createTransaction(dummyPrivateKey, selectedUtxos, recipientTotal, recipients, changeAddress, 'Mainnet');
+
+    tx.sign(hex.decode(dummyPrivateKey));
+    tx.finalize();
+
+    const txSize = tx.vsize;
+    let sentSats = new BigNumber(0);
+
+    for (let i = 0; i < tx.outputsLength; i++) {
+      const output = tx.getOutput(i);
+      sentSats = sentSats.plus(new BigNumber((output.amount ?? 0n).toString()));
+    }
+
+    const change = sentSats.minus(recipientTotal);
+    const fee = new BigNumber(txSize).times(feeRate);
```

Comment:
seems like this could be a separate function if you need to scope the variables anyway

Reply:
Good point, will extract 👍